### PR TITLE
Frontend unit tests

### PR DESCRIPTION
**Pull request by Quinn-Patrick**
All unit tests now pass. Fixed several things, including google maps api implementation and the once totally broken profile location component. Got rid of the blind guy.

### CODE REVIEW

**Comment by Quinn-Patrick** [comment]
Also, sorry about the ginormous import arrays on the spec files. They legitimately wouldn't run until I added those. If someone knows of a better way, I would happily implement it.

File: src/app/components/home-page/home-page.component.spec.ts [hunk 1]
```diff
@@ -22,4 +30,10 @@ describe('HomePageComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  class RouterStub {
+    navigateByUrl(url: string) {
```
Comment (IkenoXamos) [inline src/app/components/home-page/home-page.component.spec.ts]: What exactly is this accomplishing?

Comment (IkenoXamos) [reply src/app/components/home-page/home-page.component.spec.ts]: It seems like it is replacing the Router in the component with a stub that does nothing. But the unit test is simply testing whether the component exists.

File: src/app/components/profile-location/profile-location.component.html [hunk 2]
```diff
@@ -9,8 +9,8 @@ <h4>Location:</h4>
                     <label for="address" class="col-sm-4 col-form-label" >Street Address:</label>
                     <div class="col-sm-4">
                         <input type="text" formControlName="address2" class="form-control" id="address">
-                        <p *ngIf="addressChange.controls.streetAddress2.hasError('required') && addressChange.controls.streetAddress2.touched" class="error">Street Address required please enter one.</p>
-                        <p *ngIf="addressChange.controls.streetAddress2.hasError('pattern')" class="error">Street Address format is incorrect</p>
+                        <p *ngIf="addressChange.controls.address2.hasError('required') && addressChange.controls.address2.touched" class="error">Street Address required please enter one.</p>
```
Comment (IkenoXamos) [inline src/app/components/profile-location/profile-location.component.html:12]: Is this change needed in order for the profile-location component to work?

File: src/app/components/profile/profile.component.spec.ts [hunk 3]
```diff
@@ -10,19 +10,62 @@ import { NavbarComponent } from '../navbar/navbar.component';
...
+        ProfileCarComponent,
+        ProfileLocationComponent,
+        ProfileMembershipComponent,
+        DriverContactModalComponent,
+        DriverListComponent,
+        UserRegisterComponent,
+        HomePageComponent,
+        BsNavbarComponent,
+        ScheduleComponent,
+        ViewMyRidesComponent,
```
Comment (IkenoXamos) [inline src/app/components/profile/profile.component.spec.ts:67]: Indeed these are unfortunate. Is it possible to have several listed in the same line to conserve some vertical space? What are some alternatives that have been tried so far?

File: src/app/components/register/register.component.spec.ts [hunk 4]
```diff
@@ -6,26 +6,76 @@ import { CarRegisterComponent } from '../car-register/car-register.component';
...
+      ],
+
+      imports: [ RouterModule, RouterTestingModule, HttpClientModule, AppRoutingModule, FormsModule, ReactiveFormsModule],
+      providers: [{provide: Router, useClass: RouterStub}]
+        })
     .compileComponents();
   }));
 
+  // imports: [RouterModule, RouterTestingModule, HttpClientModule],
+  // providers: [{provide: Router, useClass: RouterStub}]
```
Comment (IkenoXamos) [inline src/app/components/register/register.component.spec.ts]: These comments can be removed.

File: src/app/services/batch-service/batch.service.spec.ts [hunk 5]
```diff
@@ -15,60 +15,93 @@ import { MyCarComponent } from 'src/app/components/my-car/my-car.component';
...
 
-//     // batchService.getAllBatches().subscribe(res => {
-//     //   response = res;
-//     // });
+    response = batchService.getAllBatches();
 
-//     expect(response).toEqual(batchResponse);
-//   });
-// });
+    expect(response).toEqual(batchResponse);
```
Comment (IkenoXamos) [inline src/app/services/batch-service/batch.service.spec.ts]: Does this unit test actually do anything?

File: src/app/services/user-service/user.service.ts [hunk 6]
```diff
@@ -55,25 +55,29 @@ export class UserService {
 	 * A GET method to retrieve one user from the database with the given id. Return a promise.
 	 * @param idParam 
 	 */
-	getUserById(idParam: number){
+	getUserById(idParam: number) 
+	{
```
Comment (IkenoXamos) [inline src/app/services/user-service/user.service.ts]: The opening curly bracket should be on the same line as the method declaration with a space between.

File: src/app/components/home-page/home-page.component.spec.ts (hunk 1 above)
Comment (Quinn-Patrick) [reply src/app/components/home-page/home-page.component.spec.ts]: You're right, that is an incorrect fix for an issue that I forgot to replace with the correct fix.

File: src/app/components/profile-location/profile-location.component.html (hunk 2 above)
Comment (Quinn-Patrick) [reply src/app/components/profile-location/profile-location.component.html:12]: Yes, this change is needed because streetAddress2 is the wrong variable name, so without that change it will say something along the lines of cannot read property hasError of undefined.

File: src/app/services/batch-service/batch.service.spec.ts (hunk 5 above)
Comment (Quinn-Patrick) [reply src/app/services/batch-service/batch.service.spec.ts]: That test verifies that the get all batches method is returning, but since it uses a spy on to control the response, it's a bit of a smoke test. I don't know why it was commented out, but it passes so I don't think it should be removed, but I admit it is a little silly.

Comment (IkenoXamos) [reply src/app/services/batch-service/batch.service.spec.ts]: Well, it is basically asserting that custom object is equal to itself.

Comment (IkenoXamos) [reply src/app/services/batch-service/batch.service.spec.ts]: It passes because we basically just have an empty unit test (barring an exception being thrown).

Comment (Quinn-Patrick) [reply src/app/services/batch-service/batch.service.spec.ts]: Okay, I'll take it out, no problem.

File: src/app/components/profile-location/profile-location.component.html (hunk 2 above)
Comment (IkenoXamos) [reply src/app/components/profile-location/profile-location.component.html:12]: Alrighty, that should be fine then.
